src/common/AndroidUtils.ts

```diff
@@ -510,45 +510,57 @@ export class AndroidUtils {
                 return AndroidUtils.emulatorHasPort(resolvedEmulator);
             })
             .then(async (port) => {
-                const resolvedPortNumer = port
+                const resolvedPortNumber = port
                     ? port
                     : await AndroidUtils.getNextAvailableAdbPort();
 
-                if (resolvedPortNumer === port) {
+                if (resolvedPortNumber === port) {
                     // already is running on a port
                     const isWritable = await AndroidUtils.isEmulatorSystemWritable(
-                        resolvedPortNumer
+                        resolvedPortNumber
                     );
 
                     if (writable === false || isWritable === true) {
                         // If we're not asked for a writable, or if it already
                         // is writable then we're done so just return its port.
-                        return Promise.resolve(resolvedPortNumer);
+                        return Promise.resolve(resolvedPortNumber);
                     }
 
                     // mismatch... shut it down and relaunch it in the right mode
-                    await AndroidUtils.stopEmulator(resolvedPortNumer);
+                    CommonUtils.updateCliAction(
+                        'emulator currently launched without the -writable-system flag so shutting it down first'
+                    );
+                    await AndroidUtils.stopEmulator(resolvedPortNumber);
                 }
 
+                CommonUtils.updateCliAction(
+                    `launching ${resolvedEmulatorName} on port ${resolvedPortNumber}${
```

Comment:
Can we conditionally add "Re-" to this, based on whether it had previously been running?

Reply:
sure, will do